src/main.rs

```diff
@@ -240,6 +271,129 @@ async fn main() -> Result<()> {
+                })
+                .collect::<Vec<_>>();
+            let failures = encrypt_result
+                .access_errs()
+                .iter()
+                .map(|edit_err| match &edit_err.user_or_group {
+                    UserOrGroup::User { id } => {
+                        format!("User: {}, Error: {}", id.id(), edit_err.err)
+                    }
+                    UserOrGroup::Group { id } => {
+                        format!("Group: {}, Error: {}", id.id(), edit_err.err)
+                    }
+                })
+                .collect::<Vec<_>>();
+            println!("Successfully encrypted file to: {:#?}", successes);
+            println!("Failed to encrypt file to: {:#?}", failures);
+            std::fs::write(&output, encrypt_result.encrypted_data())?;
+            println!("Output encrypted file to {:?}", output.display());
+        }
+        CommandLineArgs::FileDecrypt {
+            filename,
+            device_path,
+            output: maybe_output,
+        } => {
+            let output = match maybe_output {
+                // User specified output file
+                Some(desired) => desired,
+                // User did not specify output, so try to infer
+                None => {
+                    match filename.extension() {
+                        // The input file has an extension. If it's .iron, remove it
+                        Some(extension) => {
+                            // The extension is ".iron"
+                            if extension.to_os_string() == std::ffi::OsString::from("iron") {
+                                let mut output_path = filename.clone();
+                                output_path.set_extension("");
+                                std::result::Result::Ok(output_path)
+                            // The extension isn't ".iron", so we can't safely choose an output file
+                            } else {
+                                std::result::Result::Err(InitAppErr(
+                                    "No output file given, and unable to infer.".to_string(),
+                                ))
+                            }
+                        }
+                        // The filename has no extension, so we can't safely choose an output file
+                        None => std::result::Result::Err(InitAppErr(
+                            "No output file given, and unable to infer.".to_string(),
+                        )),
+                    }?
+                }
+            };
+            let sdk = initialize_sdk_from_file(&device_path).await?;
+            let file = std::fs::read(&filename)?;
+            println!("Read in file {:?}", filename);
+            let decrypt_result = sdk.document_decrypt(&file).await?;
+            std::fs::write(&output, decrypt_result.decrypted_data())?;
+            println!("Output decrypted file to {:?}", output.display());
```

Comment:
display returns something that implements Display, so you should be able to use `{}`

Reply:
Done